src/pages/Validator/AutocompleteList.js

```diff
@@ -2,36 +2,78 @@ import React from "react";
 
 export default function AutocompleteList({
   entities = {},
+  showAttributes,
   inputVal,
   onSelect,
   onCreate,
 }) {
-  // Calls onSelect to insert token into the UI
+  /*
+   * Inserts token into the UI, replacing whatever text the user already wrote 
+   * in the token
+   */
   let handleSelect = (entity) => {
     onSelect(entity);
   };
 
-  let entitiesSeen = [];
-
-  // Create a list of synonyms, filter them with the input value, and then return a token if the entity isn't already displayed
-  let shownOptions = Object.keys(entities)
-    .filter((entity) => entity.toUpperCase().startsWith(inputVal.toUpperCase()))
-    .map((entity) => {
-      if (!entitiesSeen.includes(entities[entity])) {
-        entitiesSeen.push(entities[entity])
-        return (
-          <li key={entity} onClick={handleSelect.bind(this, entities[entity])} >
-            {entities[entity]}
-          </li >
-        )
+  
+  /*
+   * Returns a list of synonym autocomplete options, filtered by the input
+   */
+  let listSynonyms = () => {
+    let entitiesSeen = [];
```

Comment:
I can't pin down exactly what `entitiesSeen` does. Could you LMK.

Reply:
We are using `entitiesSeen` to keep track of the names of the entities that have been mapped so far, so that no duplicate entity names are displayed as autocomplete options.

#### Example
The user types `fra`, which matches the synonyms `frat` and `fraternity`, which both map to the entity `CLUB`.
We don't want `CLUB` appearing twice, so we check to see if a `CLUB` list item has already been seen/created.

Reply:
I changed the name to `entitiesMatched`, if that's more clear.